From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: quh4gko8 <88831734+quh4gko8@users.noreply.github.com>
Date: Thu, 5 Feb 2026 12:52:23 +0000
Subject: [PATCH] tmp: fix crashes on certain websites when watching videos

Certain websites in desktop mode use I420 frames on its images, causing
compatibility issues as its handling is not implemented upstream, and
currrently not intended to be reached, but regressed in M145. Disable
calling NOTREACHED() for now until the issue is further investigated and
a proper fix can replace this patch.
---
 .../media/capture/desktop_capturer_android.cc       | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/content/browser/media/capture/desktop_capturer_android.cc b/content/browser/media/capture/desktop_capturer_android.cc
index 6051fb8b7b508..ad19e3aa3ae35 100644
--- a/content/browser/media/capture/desktop_capturer_android.cc
+++ b/content/browser/media/capture/desktop_capturer_android.cc
@@ -13,6 +13,10 @@
 // Must come after all headers that specialize FromJniType() / ToJniType().
 #include "content/public/android/content_jni_headers/ScreenCapture_jni.h"
 
+#if 1
+#include "base/logging.h"
+#endif
+
 namespace content {
 
 namespace {
@@ -143,6 +147,15 @@ void DesktopCapturerAndroid::OnI420FrameAvailable(
     jint unchecked_crop_top,
     jint unchecked_crop_right,
     jint unchecked_crop_bottom) {
+// BUILDFLAG(IS_ANDROID) is always true here, temporary change to avoid
+// unnecessary handling of conflicts in porting.
+#if 1
+  LOG(ERROR) << "Method is called when processing of I420 frames "
+             << "is not implemented upstream, replacing NOTREACHED() "
+             << "with debug-only check temporarily";
+  DCHECK(false);
+  return;
+#endif
   // TODO(crbug.com/352187279): Implement processing of I420 frames.
   NOTREACHED();
 }
